### ISSUE TITLE
Pipenv is issuing false conflicts when trying to install libs

### ISSUE THREAD

**Issue by frenetic** [pypa/pipenv#1745]
In a brand new environment Im trying to install `django = "==1.8.19"` and `django-redis = "==4.4.4"`.
However, `Pipenv` warns a conflict:
`Could not find a version that matches django==1.8.19,>=1.11`.

The thing is: `django-redis` at version `4.4.4` has no dependency on Django, as we can see here: https://github.com/niwinz/django-redis/blob/4.0.0/setup.py#L19
On the other hand, `django-redis` at the current release (4.9.0) and on master has a dependency on `Django 1.11`, as we can see here: https://github.com/niwinz/django-redis/blob/4.9.0/setup.py#L24

I think `Pipenv` is "looking" at the wrong versions, or is getting the wrong information, when trying to create the lock file.

Using version `11.7.4`

------------

##### Expected result

Pipenv should allow me to install Django at version 1.8.19 and Django-redis at version 4.4.4

##### Actual result

Pipenv says there is a conflict on Django versions, even though `django-redis` at version 4.4.4 has no dependency on Django.

##### Steps to replicate

In a brand new directory:
1. pipenv install django==1.8.19
2. pipenv install django-redis==4.4.4

------------

<details><summary>$ python -m pipenv.help output</summary>
Pipenv version: `'11.7.4'`

Pipenv location: `'/usr/local/lib/python3.5/dist-packages/pipenv'`

Python location: `'/usr/bin/python3'`

Other Python installations in `PATH`:

  - `2.7`: `/usr/bin/python2.7`
  - `2.7`: `/usr/bin/python2.7`
  - `3.5`: `/usr/bin/python3.5m`
  - `3.5`: `/usr/bin/python3.5`

  - `2.7.12`: `/usr/bin/python`
  - `2.7.12`: `/usr/bin/python2`
  - `3.5.2`: `/usr/bin/python3`

PEP 508 Information:

```
{'implementation_name': 'cpython',
 'implementation_version': '3.5.2',
 'os_name': 'posix',
 'platform_machine': 'x86_64',
 'platform_python_implementation': 'CPython',
 'platform_release': '4.13.0-36-generic',
 'platform_system': 'Linux',
 'platform_version': '#40~16.04.1-Ubuntu SMP Fri Feb 16 23:25:58 UTC 2018',
 'python_full_version': '3.5.2',
 'python_version': '3.5',
 'sys_platform': 'linux'}
```

System environment variables:

  - `BYOBU_ULIMIT`
  - `BYOBU_CHARMAP`
  - `BYOBU_READLINK`
  - `LC_CTYPE`
  - `DESKTOP_SESSION`
  - `XDG_CURRENT_DESKTOP`
  - `LC_TELEPHONE`
  - `LSCOLORS`
  - `EVENT_NOEPOLL`
  - `GTK2_MODULES`
  - `QT_QPA_PLATFORMTHEME`
  - `QT_ACCESSIBILITY`
  - `IM_CONFIG_PHASE`
  - `XDG_DATA_DIRS`
  - `SHLVL`
  - `QT4_IM_MODULE`
  - `GDMSESSION`
  - `GTK_MODULES`
  - `BYOBU_SED`
  - `SESSIONTYPE`
  - `NVM_BIN`
  - `LC_NUMERIC`
  - `BYOBU_TIME`
  - `LC_ADDRESS`
  - `BYOBU_PREFIX`
  - `XDG_VTNR`
  - `LESS`
  - `SESSION`
  - `UNITY_HAS_3D_SUPPORT`
  - `SSH_AUTH_SOCK`
  - `SESSION_MANAGER`
  - `XDG_SEAT`
  - `WINDOWID`
  - `QT_LINUX_ACCESSIBILITY_ALWAYS_ON`
  - `LANGUAGE`
  - `BYOBU_PYTHON`
  - `LC_PAPER`
  - `BYOBU_LIGHT`
  - `BYOBU_DARK`
  - `XDG_MENU_PREFIX`
  - `DEFAULTS_PATH`
  - `BYOBU_BACKEND`
  - `PATH`
  - `BYOBU_RUN_DIR`
  - `VTE_VERSION`
  - `XAUTHORITY`
  - `BYOBU_CONFIG_DIR`
  - `GNOME_KEYRING_CONTROL`
  - `UNITY_DEFAULT_PROFILE`
  - `BYOBU_PAGER`
  - `XDG_RUNTIME_DIR`
  - `XDG_SESSION_PATH`
  - `LC_MONETARY`
  - `XDG_GREETER_DATA_DIR`
  - `LC_MEASUREMENT`
  - `XDG_SESSION_TYPE`
  - `_`
  - `NVM_CD_FLAGS`
  - `SHELL`
  - `LS_COLORS`
  - `LOGNAME`
  - `GDM_LANG`
  - `INSTANCE`
  - `XDG_SESSION_DESKTOP`
  - `PYTHONUNBUFFERED`
  - `LC_IDENTIFICATION`
  - `GNOME_KEYRING_PID`
  - `JOB`
  - `NVM_DIR`
  - `XMODIFIERS`
  - `TMUX`
  - `UPSTART_INSTANCE`
  - `GTK_IM_MODULE`
  - `TERM`
  - `DISPLAY`
  - `GPG_AGENT_INFO`
  - `HOME`
  - `COMPIZ_CONFIG_PROFILE`
  - `UPSTART_EVENTS`
  - `OLDPWD`
  - `PWD`
  - `BYOBU_WINDOW_NAME`
  - `QT_IM_MODULE`
  - `DBUS_SESSION_BUS_ADDRESS`
  - `USER`
  - `ZSH`
  - `PIP_PYTHON_PATH`
  - `BYOBU_DATE`
  - `TMUX_PANE`
  - `XDG_SEAT_PATH`
  - `BYOBU_HIGHLIGHT`
  - `XDG_SESSION_ID`
  - `GNOME_DESKTOP_SESSION_ID`
  - `PAGER`
  - `LANG`
  - `LC_TIME`
  - `XDG_CONFIG_DIRS`
  - `COMPIZ_BIN_PATH`
  - `BYOBU_DISTRO`
  - `BYOBU_ACCENT`
  - `MANDATORY_PATH`
  - `BYOBU_TTY`
  - `LC_NAME`
  - `UPSTART_SESSION`
  - `UPSTART_JOB`
  - `CLUTTER_IM_MODULE`

Pipenv–specific environment variables:

Debug–specific environment variables:

  - `PATH`: `/home/frenetic/.nvm/versions/node/v8.9.4/bin:/home/frenetic/bin:/home/frenetic/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin`
  - `SHELL`: `/usr/bin/zsh`
  - `LANG`: `en_US.UTF-8`
  - `PWD`: `/home/frenetic/Projects/tests/tests-api`

---------------------------

Contents of `Pipfile` ('/home/frenetic/Projects/tests/tests-api/Pipfile'):

```toml
[[source]]

name = "pypi"
url = "https://pypi.python.org/simple"
verify_ssl = true


[packages]

django = "==1.8.19"
django-redis = "==4.4.4"

[dev-packages]


[requires]

python_version = "2.7"

```

**Comment by gsemet** [comment]
reproduced. Wonder if it is not in the same scope than #1732 (transitive dependency resolution problem)

**Comment by jtratner** [comment]
FYI - django-redis==4.9.0 (latest) *does* require Django>=1.11

**Comment by gsemet** [comment]
indeed, so i think the behavior is normal

**Comment by kennethreitz** [comment]
![](http://share.kennethreitz.org/190x0u3T2a1L/Screen%20Shot%202018-03-14%20at%206.42.31%20PM.png)

**Comment by kennethreitz** [comment]
![](http://share.kennethreitz.org/1V0h3k3s1T1Q/Screen%20Shot%202018-03-14%20at%206.49.25%20PM.png)

the first thing it does is look for django

**Comment by kennethreitz** [comment]
`notpip.index:Using version 4.9.0 (newest of versions: 1.1rc1, 1.2, 1.2.1, 1.3, 1.3.1, 1.4, 1.4.1, 1.4.3, 1.4.4, 1.4.5, 1.5, 1.5.1, 1.5.2, 2.0rc1, 2.0rc2, 2.0rc3, 2.1, 2.2, 2.2.2, 3.0, 3.1, 3.1.1, 3.1.2, 3.1.3, 3.1.4, 3.1.5, 3.1.6, 3.1.7, 3.2, 3.3, 3.4.0, 3.5.0, 3.5.1, 3.5.2, 3.6.0, 3.6.1, 3.6.2, 3.7.0, 3.7.1, 3.7.2, 3.8.0, 3.8.1, 3.8.2, 3.8.3, 3.8.4, 4.0.0, 4.1.0, 4.2.0, 4.3.0, 4.4.0, 4.4.1, 4.4.2, 4.4.3, 4.4.4, 4.5.0, 4.6.0, 4.7.0, 4.8.0, 4.8.0, 4.9.0, 4.9.0)`

**Comment by kennethreitz** [comment]
got it!

**Comment by kennethreitz** [comment]
use 11.8.0 :)

**Comment by frenetic** [comment]
Im sorry to get back to this, but this bug was not fixed.

Using the newest `11.8.0` version I created a new directory and did a simple `pipenv install django-redis==4.4.4`. It is asking for the latest Django version, even though `django-redis` at version `4.4.4` does not declare this requirement.

![image](https://user-images.githubusercontent.com/122976/37472443-df650086-284a-11e8-8918-2eb9baba161e.png)

The weird part is, even though it checks for Django, it is not listed as its dependency on `pipenv graph`.

![image](https://user-images.githubusercontent.com/122976/37472520-0d6aa13e-284b-11e8-8b6b-e7541ad9fbbe.png)

Finally, in this same directory that I installed `django-redis==4.4.4`, if I try to install Django in a version < 1.11, I still get the dependency error.

![image](https://user-images.githubusercontent.com/122976/37472646-5ece28fc-284b-11e8-974f-6550e86638ea.png)
